_gsocblogs/2022/blog_EtherpadPlugin_MohammadWarid.md

```diff
@@ -46,7 +46,33 @@ This implementation served well for the project which meant that the metadata ca
 
 Currently, I am investigating the WOPI server functionality for receiving the metadata and handling it, after which, I will move towards testing the complete workflow.
 
+# Post Midterm Period
+
+## Support for multi-user collaboration
+The main goal for the project was an in-app workflow, targeted to onboard new users over the etherpad each possessing a unique ID. This was achieved by fetching query parameters from a custom `/setEFSSMetadata` endpoint, thereby passing successful or error based responses to the wopiserver, which then allocated a new url to serve the etherpad user. 
+The endpoint was protected against validation checks for bogus inputs like invalid `padID` and `API` key.
+
+## Debouncing padUpdate event
+The `padUpdate` event in etherpad is supposed to fire on every new change to the etherpad. This is more noticeable when dealing with every keystroke event. To lower the amount of `pending 202` requests at the wopiserver’s end, the `padUpdate` event was debounced by a factor of 3000 milliseconds. After this implementation, only the latest updated change was taken into account. 
```

Comment:
Let's simplify `To lower the amount of pending requests` into `To lower the amount of requests` - As you know the wopiserver is performing its own "debouncing" and that's the reason it responds with 202.